x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TrainedModel.java

```diff
@@ -6,21 +6,14 @@
 package org.elasticsearch.xpack.core.ml.inference.trainedmodel;
 
 import org.apache.lucene.util.Accountable;
-import org.elasticsearch.common.Nullable;
 import org.elasticsearch.common.io.stream.NamedWriteable;
 import org.elasticsearch.xpack.core.ml.inference.results.InferenceResults;
 import org.elasticsearch.xpack.core.ml.utils.NamedXContentObject;
 
-import java.util.List;
 import java.util.Map;
 
 public interface TrainedModel extends NamedXContentObject, NamedWriteable, Accountable {
 
-    /**
-     * @return List of featureNames expected by the model. In the order that they are expected
-     */
-    List<String> getFeatureNames();
```

Comment:
Both these methods are just not called anywhere? Is that why they are removed? If so 👍

Reply:
Yes, I am cleaning up the interface. I added a bunch of stuff that I thought "all" models would need, and it turns out that that is not the case.